Assets/Scripts/ThrusterManager.cs

```diff
@@ -4,22 +4,58 @@ public class ThrusterManager : MonoBehaviour
 {
     private const KeyCode EngageKey = KeyCode.Space;
     
+    public float thrustCapacityEffectInteractionDelay = 5f;
     public float power = 10f;
     public float maxCapacity = 10f; // in seconds
     public AudioSource thrusterEngineSfx, thrusterEmptySfx;
     
     private bool engaged;
     private float capacity; // in seconds
     private UIManager ui;
+    private float lastThrustCapacityEffect;
 
     private void Start()
     {
+        lastThrustCapacityEffect = -thrustCapacityEffectInteractionDelay;
         engaged = false;
         capacity = maxCapacity;
         ui = FindObjectOfType<UIManager>();
         ui.fuelGauge.SetVal(maxCapacity, maxCapacity);
     }
 
+    private void FillFuelTank(float amount)
+    {
+        SetCapacity(capacity + amount);
+    }
+
+    private void EmptyFuelTank(float amount)
+    {
+        SetCapacity(capacity - amount);
+    }
+
+    private void SetCapacity(float newCapacity)
+    {
+        capacity = Mathf.Clamp(newCapacity, 0, maxCapacity);
+        ui.fuelGauge.SetVal(capacity, maxCapacity);
+    }
+
+    public void ProcessThrustCapacityEffect(GameObject healthEffectObj)
+    {
+        if (Time.fixedTime - lastThrustCapacityEffect <= thrustCapacityEffectInteractionDelay) return;
```

Comment:
why `Time.fixedTime` instead of `Time.time`?

Reply:
This came from the SpaceSuitManager, so idk tbh... I think it has to do with it being done in a collision?